api/fake_agent_server_test.go

```diff
@@ -0,0 +1,146 @@
+package api_test
+
+import (
+	"context"
+	"log/slog"
+	"testing"
+
+	"github.com/buildkite/agent-stack-k8s/v2/api"
+	"github.com/buildkite/stacksapi"
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestFakeAgentServer_DefaultBehavior(t *testing.T) {
+	ctx := context.Background()
+
+	// Create fake server
+	server := api.NewFakeAgentServer()
+	defer server.Close()
+
+	// Create REAL AgentClient pointing to fake server
+	client, err := api.NewAgentClient(ctx, api.AgentClientOpts{
+		Token:    "fake-token",
+		Endpoint: server.URL(),
+		StackID:  "test-stack",
+		Logger:   slog.Default(),
+	})
+	if err != nil {
+		t.Fatalf("NewAgentClient() error = %v", err)
+	}
+
+	jobIDs := []string{"job-1", "job-2", "job-3"}
+	result, retryAfter, err := client.ReserveJobs(ctx, jobIDs)
+
+	want := &stacksapi.BatchReserveJobsResponse{
+		Reserved:    jobIDs,
+		NotReserved: []string{},
+	}
+
+	if err != nil {
+		t.Errorf("ReserveJobs() error = %v, want nil", err)
+	}
+	if retryAfter != 0 {
+		t.Errorf("ReserveJobs() retryAfter = %v, want 0", retryAfter)
+	}
+
+	if diff := cmp.Diff(result, want); diff != "" {
+		t.Errorf("ReserveJobs(ctx, %q) diff (-got +want):\n%s", jobIDs, diff)
+	}
```

Comment:
1. Don't separate `..., err := ...` from `if err != nil`
2. Include the argument to the function being called (`ReserveJobs`) in output

```suggestion
	result, retryAfter, err := client.ReserveJobs(ctx, jobIDs)
	if err != nil {
		t.Errorf("ReserveJobs(%q) error = %v, want nil", jobIDs, err)
	}
	if retryAfter != 0 {
		t.Errorf("ReserveJobs(%q) retryAfter = %v, want 0", jobIDs, retryAfter)
	}
	
	want := &stacksapi.BatchReserveJobsResponse{
		Reserved:    jobIDs,
		NotReserved: []string{},
	}
	if diff := cmp.Diff(result, want); diff != "" {
		t.Errorf("ReserveJobs(ctx, %q) diff (-got +want):\n%s", jobIDs, diff)
	}
```

Reply:
Don't forget to apply throughout